bin/cancel-pending-tx.js

```diff
@@ -0,0 +1,172 @@
+  console.log('https://filfox.info/en/message/%s', tx)
+  console.log('Note: we have access to 50 oldest messages only.')
+  for (const { cid, createTimestamp } of messages) {
+    console.log(' %s (created at %s)', cid, new Date(createTimestamp * 1000).toISOString())
+  }
+  process.exit(1)
+}
+console.log('TX message:', txDetails)
+
+/*
+https://filecoinproject.slack.com/archives/C0179RNEMU4/p1607269156412600?thread_ts=1606987775.021900&cid=C0179RNEMU4
+while you can not cancel a message, you have an option to replace it with a different one, in this
+case you would replace it with the cheapest possible one ( a self-send of 0 ) in order to "convince"
+the network to take the new message all you have to do is to increase your --gas-premium by 25.2%
+from the previous value ( technically it is 25% + 1 attofil, but 25.2 is easier )
+
+you also must LOWER your gas-limit to match the new gas requirements of the message you are sending:
+if you do not do so, the network will penalize you for lying ( you said this will cost X gas but
+used 1/10th of X to execute: no good )
+
+all in all the steps to replace a message (currently to be fixed soon) are:
+- you go to some explorer, find a recent send message, that just landed on chain
+- you note what is its gas-used (RECENTGU) and gas-feecap (RECENTFC)
+- you find your own message you want to replace and find its current gas-premium (OLDGP)
+- then you do magic:
+
+lotus send
+  --gas-feecap $RECENTFC-from-above \
+  --gas-limit $RECENTGU*1.1 \
+  --gas-premium $OLDGP*1.252  \
+  --from {{address-you-are-unblocking}} \
+  --nonce {{nonce-you-are-replacing}} \
+  {{address-you-are-unblocking-as-destination}} \
+  0
+
+that should be all, and should not cost you much at all ( we just cleared ~80 messages in the thread
+above by paying ~0.05 fil
+ */
+
+const recentSendMessage = await getRecentSendMessage()
+console.log('Calculating gas fees from the recent Send message %s (created at %s)',
+  recentSendMessage.cid,
+  new Date(recentSendMessage.timestamp * 1000).toISOString()
+)
+
+const gasUsed = recentSendMessage.receipt.gasUsed
+const gasFeeCap = Number(recentSendMessage.gasFeeCap)
+const oldGasPremium = Number(txDetails.gasPremium)
+const nonce = txDetails.nonce
+
+console.log('SENDING THE REPLACEMENT TRANSACTION')
+const replacementTx = await signer.sendTransaction({
+  to: walletDelegatedAddress,
+  value: 0,
+  nonce,
+  gasLimit: Math.ceil(gasUsed * 1.1),
+  maxFeePerGas: gasFeeCap,
+  maxPriorityFeePerGas: Math.ceil(oldGasPremium * 1.252)
+})
+console.log('Waiting for the transaction receipt:', replacementTx.hash)
+const receipt = await replacementTx.wait()
+console.log('TX status:', receipt?.status)
+
+/**
+ * @param {string} f4addr
+ * @returns 1000 oldest messages
+ */
+async function getMessagesInMempool (f4addr) {
+  const res = await fetch(`https://filfox.info/api/v1/message/mempool/filtered-list?address=${f4addr}&pageSize=1000`)
```

Comment:
I doubt 1000 is good enough, can we increase this or use pagination, until we found the right one?